src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -448,29 +448,11 @@ private BoundStatement BindGoto(GotoStatementSyntax node, DiagnosticBag diagnost
 
         private BoundStatement BindLocalFunctionStatement(LocalFunctionStatementSyntax node, DiagnosticBag diagnostics)
         {
-            var binder = GetBinder(node);
-            Debug.Assert(binder != null);
-
-            return binder.BindLocalFunctionStatementParts(node, diagnostics);
-        }
-
-        private BoundStatement BindLocalFunctionStatementParts(LocalFunctionStatementSyntax node, DiagnosticBag diagnostics)
-        {
             // already defined symbol in containing block
             var localSymbol = this.LookupLocalFunction(node.Identifier);
 
-            var hasErrors = false;
-
-            // In error scenarios with misplaced code, it is possible we can't bind the local declaration.
-            // This occurs through the semantic model.  In that case concoct a plausible result.
-            if (localSymbol == null)
-            {
-                localSymbol = new LocalFunctionSymbol(this, this.ContainingMemberOrLambda, node);
-            }
-            else
-            {
-                hasErrors |= this.ValidateDeclarationNameConflictsInScope(localSymbol, diagnostics);
-            }
+            var binder = localSymbol.ScopeBinder;
```

Comment:
>var binder = localSymbol.ScopeBinder [](start = 12, length = 36)

Consider renaming to scopeBinder or inlining the local so that no one would try to use this binder for binding something.